### PR TITLE
help generate_docs | flatten crashes nushell

### LINKED ISSUE #3094: help generate_docs crashes nushell
**Describe the bug**
If someone else could confirm they are seeing the same behavior that would be good...

**To Reproduce**
1. help generate_docs

### PR DESCRIPTION
**Pull request by stormasm**
this fixes the case where the user would type

```
help generate_docs | flatten
```

and nushell would crash

closes #3094 

fix case where parent_name was {nu, term} and possibly others in the future by doing an extra test first to see if if the *parent_name key actually exists in cmap...

### CODE REVIEW

File: crates/nu-engine/src/documentation.rs [hunk 1]
```diff
@@ -67,10 +67,12 @@ pub fn generate_docs(scope: &Scope) -> Value {
...
-            let sub_names = cmap
-                .get_mut(*parent_name)
-                .expect("Expected a entry for parent");
-            sub_names.push(name.to_owned());
+            if cmap.contains_key(*parent_name) {
+                let sub_names = cmap
+                    .get_mut(*parent_name)
+                    .expect("Expected a entry for parent");
+                sub_names.push(name.to_owned());
+            }
```
Comment (jonathandturner) [inline crates/nu-engine/src/documentation.rs:75]: Can you create a test case for this, too?

Comment (stormasm) [reply crates/nu-engine/src/documentation.rs:75]: yes, sounds good...

Comment (stormasm) [reply crates/nu-engine/src/documentation.rs:75]: @jonathandturner ok the tests now cover showing both

```rust
help commands
```

```rust
help generate_docs
```
is working...

**Comment by jonathandturner** [comment]
Looks good!

**Pull request merged**